File: gitlfs-client/src/main/java/ru/bozaro/gitlfs/client/io/StringStreamProvider.java
package ru.bozaro.gitlfs.client.io;

import javax.annotation.Nonnull;
import java.nio.charset.StandardCharsets;

/**
 * Create stream from string.
 *
 * @author Artem V. Navrotskiy
 */
public class StringStreamProvider extends ByteArrayStreamProvider {
  public StringStreamProvider(@Nonnull String data) {
    super(data.getBytes(StandardCharsets.UTF_8));
  }
}

File: gitlfs-client/src/main/kotlin/ru/bozaro/gitlfs/client/io/StringStreamProvider.kt
package ru.bozaro.gitlfs.client.io

import java.nio.charset.StandardCharsets

/**
 * Create stream from string.
 *
 * @author Artem V. Navrotskiy
 */
class StringStreamProvider(data: String) : ByteArrayStreamProvider(data.toByteArray(StandardCharsets.UTF_8))
